src/Winton.Extensions.Configuration.Consul/Parsers/JsonConfigurationParser.cs

```diff
@@ -14,14 +14,31 @@ namespace Winton.Extensions.Configuration.Consul.Parsers
     /// </summary>
     public sealed class JsonConfigurationParser : IConfigurationParser
     {
-        /// <inheritdoc />
+        private sealed class JsonStreamParser : JsonStreamConfigurationProvider
+        {
+            private JsonStreamParser(JsonStreamConfigurationSource source)
+                : base(source)
+            {
+            }
+
+            internal static IDictionary<string, string> Parse(Stream stream)
+            {
+                var provider = new JsonStreamParser(new JsonStreamConfigurationSource { Stream = stream });
+                provider.Load();
+                return provider.Data;
+            }
+        }
+
+        /// <summary>
+        /// Parse the <see cref="Stream" /> into a dictionary.
+        /// </summary>
+        /// <param name="stream">The stream to parse.</param>
+        /// <returns>A dictionary representing the configuration in a flattened form.</returns>
+#pragma warning disable SA1201 // Elements should appear in the correct order
```

Comment:
Rather than disable and re-enable the warning I think you can just move this method above the `private sealed class JsonStreamParser` and it will work.

Reply:
agree .